src/ui/public/courier/fetch/request/segmented.js

```diff
@@ -81,10 +83,12 @@ define(function (require) {
         // request, making sure the first request returns faster.
         var remainingSegments = self._maxSegments - self._segments.length;
         var indexCount = Math.max(1, Math.floor(self._queue.length / remainingSegments));
-        params.index = self._active = self._queue.splice(0, indexCount);
 
-        if (self._desiredSize !== false) {
-          params.body.size = Math.max(self._desiredSize - self._hitsReceived, 0);
+        var indices = self._active = self._queue.splice(0, indexCount);
+        params.index = _.pluck(indices, 'index');
+
+        if (self._desiredSize) {
```

Comment:
The same question applies to all conditional checks of `desiredSize`, but I'm not going to spam the PR with each.